### DISCUSSION TITLE
Checkboxes get re-checked when I uncheck them, why? [SOLVED]

### DISCUSSION THREAD

**Discussion by berwynn** [livewire/livewire#2669] [1. Help]
First of all, sorry if this question has been asked, I have been looking around for a while but I couldn't find anything that would be specific to my issue. Second, I apologize for the crappy title, but essentially if there are multiple checkboxes on a page and I go to uncheck one that already is in a checked state, the checkbox will uncheck and about half a second later re-checks itself.

Here is the front-end snippet: 
```
<div class=" mt-2 col-md-6">
        <h3 class="col-md-12 mt-3">Available Products</h3>
        @foreach($products->where('product_category_id', $categoryID) as $index => $product)
        <div class="col-md-12 mt-2">
            <div class="col-md-10 px-0">
                <input disabled class="form-input col-md-12" type="text" value="{{$product->name}}">
            </div>
            <div class="col-md-2">
                <input wire:click="toggleProduct({{$product->id}})" class="mt-1" type="checkbox" value="{{$product->id}}" @if($company->products->pluck('id')->contains($product->id)) checked @endif>
            </div>
        </div>
        @endforeach
</div>
```

The problem is with this line: 
`<input wire:click="toggleProduct({{$product->id}})" class="mt-1" type="checkbox" value="{{$product->id}}" @if($company->products->pluck('id')->contains($product->id)) checked @endif>`

The backend function "toggleProduct()" looks like this: 
```
public function toggleProduct($id) {
        if($this->company->products->contains($id)){
            $this->company->products()->detach($id);
            session()->flash('snack-success', 'Product successfully detached');
        }
        else {
            $this->company->products()->attach($id);
            session()->flash('snack-success', 'Product successfully attached');
        }
        $this->dispatchBrowserEvent('success-shown');
    }
```
I am not sure what I am doing wrong, or if I need to create a weird array of checkboxes then have that populate with the products during mount? Though that seems unnecessary to me. 

Any help and advice would be appreciated, thank you!

**Comment by joshhanley** [comment]
@WS243593 you may want to check out this issue as I believe it is related https://github.com/livewire/livewire/issues/788

**Reply by berwynn** [reply]
Finally fixed it, gonna comment my solution. #788's answer by @JayBizzle did it for me. Thank you so much for your help!

**Comment by berwynn** [comment]
Here is my solution and reasoning:

Correct me if I am wrong, but the way I set it up above, especially front end on this line:
```
<input wire:click="toggleProduct({{$product->id}})" class="mt-1" type="checkbox" value="{{$product->id}}" @if($company->products->pluck('id')->contains($product->id)) checked @endif>
```
I didn't hook this up to any livewire model, so it had no way of tracking the checkboxes, again I could be wrong here. So going by what Jay said on #788, I have added a separate array that was for the purpose of tracking the checkboxes like so:
```
public $products = [];
public $productIDs = [];
```
These then get populated like this:
```
$this->products = Product::all();
$this->productIDs = array_map('strval', $this->company->products()->pluck('id')->all());
```
So since I don't know what company will have what products I have to pull them all down, then I can track the existing ones by the 'productIDs' variable.

Then finally, hooked the above mentioned line with the correct wire model:
```
<input wire:model="productIDs" wire:click="toggleProduct({{$product->id}})" class="mt-1" type="checkbox" value="{{$product->id}}" @if($company->products->pluck('id')->contains($product->id)) checked @endif>
```

I hope if anyone else comes across this issue, that this may serve you well!
